sdk/src/test/tutorial/java/com/finbourne/lusid/tutorials/ibor/ReferencePortfolio.java

```diff
@@ -0,0 +1,149 @@
+package com.finbourne.lusid.tutorials.ibor;
+
+import com.finbourne.features.LusidFeature;
+import com.finbourne.lusid.ApiClient;
+import com.finbourne.lusid.ApiException;
+import com.finbourne.lusid.api.InstrumentsApi;
+import com.finbourne.lusid.api.PortfoliosApi;
+import com.finbourne.lusid.api.ReferencePortfolioApi;
+import com.finbourne.lusid.model.*;
+import com.finbourne.lusid.utilities.*;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class ReferencePortfolio {
+    private static List<String> instrumentIds;
+    private static PortfoliosApi portfoliosApi;
+    private static ReferencePortfolioApi referencePortfoliosApi;
+
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        ApiConfiguration apiConfiguration = new ApiConfigurationBuilder().build(CredentialsSource.credentialsFile);
+        ApiClient apiClient = new ApiClientBuilder().build(apiConfiguration);
+
+        portfoliosApi = new PortfoliosApi(apiClient);
+        referencePortfoliosApi = new ReferencePortfolioApi(apiClient);
+        InstrumentsApi instrumentsApi = new InstrumentsApi(apiClient);
+
+        InstrumentLoader instrumentLoader = new InstrumentLoader(instrumentsApi);
+        instrumentIds = instrumentLoader.loadInstruments();
+
+        Collections.sort(instrumentIds);
+    }
+
+    @Test
+    @LusidFeature("F39")
+    public void create_reference_portfolio() throws ApiException {
+
+        // Declare name, code of the portfolio to create
+        String F39PortfolioCode = "F39J_ReferencePortfolioCode";
+        String F39PortfolioName = "F39J_Reference Portfolio name";
+
+        // Generate request using CreateReferencePortfolioRequest model
+        CreateReferencePortfolioRequest request = new CreateReferencePortfolioRequest()
+                .code(F39PortfolioCode)
+                .displayName(F39PortfolioName);
+
+        // Make the request using the reference portfolio API
+        Portfolio portfolio = referencePortfoliosApi.createReferencePortfolio(TestDataUtilities.TutorialScope, request);
+
+        // Test that result portfolio code is the same as request code
+        assertEquals(request.getCode(), portfolio.getId().getCode());
+
+        // Delete portfolio once the test is completed
+        portfoliosApi.deletePortfolio(TestDataUtilities.TutorialScope, F39PortfolioCode);
+    }
+
+    @Test
+    @LusidFeature("F40")
+    public void upsert_reference_portfolio_constituents() throws ApiException {
+
+        // Declare name, code, effective date of the portfolio to create
+        String F40PortfolioCode = "F40J_ReferencePortfolioCode";
+        String F40PortfolioName = "F40J_Reference Portfolio name";
```

Comment:
same thing here with F40, I'd say remove this as the feature number might change